php/commands/src/CLI_Alias_Command.php

```diff
@@ -0,0 +1,335 @@
+ *     Success: Updated 'prod' alias.
+ *
+ *     # Delete alias.
+ *     $ wp cli alias delete @prod
+ *     Success: Deleted '@prod' alias.
+ *
+ * @package wp-cli
+ * @when before_wp_load
+ */
+class CLI_Alias_Command extends WP_CLI_Command {
+
+	/**
+	 * List available WP-CLI aliases.
+	 *
+	 * Aliases are shorthand references to WordPress installs. For instance,
+	 * `@dev` could refer to a development install and `@prod` could refer to
+	 * a production install. This command gives you visibility in what
+	 * registered aliases you have available.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: yaml
+	 * options:
+	 *   - yaml
+	 *   - json
+	 *   - var_export
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # List all available aliases.
+	 *     $ wp cli alias list
+	 *     ---
+	 *     @all: Run command against every registered alias.
+	 *     @prod:
+	 *       ssh: runcommand@runcommand.io~/webapps/production
+	 *     @dev:
+	 *       ssh: vagrant@192.168.50.10/srv/www/runcommand.dev
+	 *     @both:
+	 *       - @prod
+	 *       - @dev
+	 *
+	 * @subcommand list
+	 */
+	public function list_( $args, $assoc_args ) {
+		WP_CLI::print_value( WP_CLI::get_runner()->aliases, $assoc_args );
+	}
+
+	/**
+	 * Gets the value for an alias.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <key>
+	 * : Key for the alias.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get alias.
+	 *     $ wp cli alias get @prod
+	 *     ssh: dev@somedeve.env:12345/home/dev/
+	 *
+	 * @subcommand get
+	 */
+	public function get( $args, $assoc_args ) {
+		list( $alias ) = $args;
+
+		$aliases = WP_CLI::get_runner()->aliases;
+
+		if ( ! empty( $aliases[ $alias ] ) ) {
+			foreach ( $aliases[ $alias ] as $key => $value ) {
+				WP_CLI::log( sprintf( ' %1$s: %2$s', $key, $value ) );
+			}
+		} else {
+			WP_CLI::error( sprintf( 'No alias found with key \'%s\'.', $alias ) );
+		}
+	}
+
+	/**
+	 * Creates an alias.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <key>
+	 * : Key for the alias.
+	 *
+	 * <login>
+	 * : SSH Login.
+	 *
+	 * <wp_path>
+	 * : WordPress install path.
+	 *
+	 * [--config=<config>]
+	 * : Config file to be considered for operations.
+	 * ---
+	 * default: global
+	 * options:
+	 *   - global
+	 *   - project
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Add alias to global config.
+	 *     $ wp cli alias add @prod login@host /path/to/wordpress/install/
+	 *     Success: Added '@prod' alias.
+	 *
+	 *     # Add alias to project config.
+	 *     $ wp cli alias add @prod login@host /path/to/wordpress/install/ --config=project
+	 *     Success: Added '@prod' alias.
+	 *
+	 * @subcommand add
+	 */
+	public function add( $args, $assoc_args ) {
+
+		$config = ( ! empty( $assoc_args['config'] ) ? $assoc_args['config'] : 'global' );
+
+		list( $config_path, $aliases ) = $this->_get_aliases_data( $config, '' );
+
+		$this->_validate_config_file( $config_path );
+
+		list( $alias, $login, $wp_path ) = $args;
```

Comment:
I don't think this correctly works as syntax format for all use cases. As the documentation states:
```
# An alias can include 'user', 'url', 'path', 'ssh', or 'http'
```
So we would need to enable users to add any of these fields to the alias, not just the `ssh` one.

I'd recommend going with the following signature instead:
```
wp cli alias add [--user=<user>] [--url=<url>] [--path=<path>] [--ssh=<ssh>] [--http=<http>] [--grouping=<comma-separated list of aliases>]
```

For the `--grouping`, this would mean that the following command:
```
wp cli alias add @someservers --grouping=first,second,third
```
would create the following config file:
```
@someservers
  - @first
  - @second
  - @third
```

Reply:
@schlessera I've done changes for other suggestions.

https://make.wordpress.org/cli/handbook/config/
```
@staging:
    ssh: wpcli@staging.wp-cli.org
    user: wpcli
    path: /srv/www/staging.wp-cli.org
```

I am not sure about other params in the doc for `http` and `url`. Will they be added like rest of them?

And if `--grouping` is provided we will ignore the rest of the input?

Reply:
Yes, `http` and `url` are added just like the others.

`--grouping` only works alone. If `--grouping` is provided, any other properties should throw an error.